### PR TITLE
LibGUI: Add some checks to make ctrl backspacing work for tabs and single chars

### PR DESCRIPTION
**Pull request by lstrsrt**
Previously, trying to delete tabs by ctrl backspacing on an otherwise empty line resulted in the program crashing (tested in Text Editor, Hack Studio and a command palette). Trying to ctrl backspace a single char resulted in a hang from a neverending loop in TextDocument::first_word_break_before().

This pull request additionally updates TextPosition to use 64-bit sentinels from NumericLimits.h (size_t is 64-bits now).

### CODE REVIEW

File: Userland/Libraries/LibGUI/TextDocument.cpp [hunk 1]
```diff
@@ -765,6 +766,8 @@ TextPosition TextDocument::first_word_break_before(TextPosition const& position,
 
             target.set_column(*index);
             break;
+        } else {
+            break;
```
Comment (konradekk) [inline Userland/Libraries/LibGUI/TextDocument.cpp:770]: it does not look bad in such form:
```suggestion
        }
        break;
```

**Comment by awesomekling** [comment]
> Hello!
> 
> One or more of the commit messages in this PR do not match the SerenityOS [code submission policy](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md?rgh-link-date=2023-04-27T15%3A43%3A38Z#code-submission-policy), please check the `lint_commits` CI job for more details on which commits were flagged and why. Please do not close this PR and open another, instead modify your commit message(s) with [git commit --amend](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message) and force push those changes to update this PR.

Please update the PR to conform to the [contributing guidelines](https://github.com/SerenityOS/serenity/blob/a268dcb1e25541a571c43e27c2ed3de1bdcd7189/CONTRIBUTING.md)

Specifically:
- Squash the fixup.
- Write a proper commit message.

**Comment by lstrsrt** [comment]
Attempted to reword the commit message, but I guess it's still too long and the whole history is now messed up. No idea how to squash these.

**Comment by AtkinsSJ** [comment]
It's definitely fixable. This video is helpful: https://m.youtube.com/watch?v=ElRzTuYln0M

Use an interactive rebase to drop the merge commit, and amend the first commit's title to be within 72 characters.

**Pull request closed without merging**